lib/galaxy/datatypes/converters/tabular_csv.py

```diff
@@ -7,7 +7,17 @@
 
 import argparse
 import csv
+import sys
 
+from contextlib import contextmanager
+
+@contextmanager
+def compat_open(file, mode):
+    if sys.version_info[0] > 2:
+        fh = open(file, mode, newline='')
+    else:
+        fh = open(file, mode + 'b')
+    yield fh
```

Comment:
```suggestion
    try:
        yield fh
    finally:
        fh.close()
```
The file needs to be closed upon exit from the context manager; I also think the yield statement needs to be wrapped in a try/finally: if an exception is raised in the `with` block, it will be re-raised in the `yield` statement. Without error handling whatever follows the yield statement (`fh.close()`) won't be executed.

I'm not sure whether we should worry about not wrapping the code that actually opens the file. Wrapping it in the same try block seems wrong, but multiple try blocks seem like overkill here. Depends on what we want the system to do if the file doesn't open here; and I don't know.

Reply:
Correction: see comment by @nsoranzo